### PR TITLE
add logic to handle aws token for sso accounts

### LINKED ISSUE #141: Migrate CircleCI pipeline to Github actions
# Summary
 Migrate the BitOps CI pipeline which is currently in CircleCI to Github actions.

The same features that are relied on today need to be present after the migration. 
The CircleCI template can be found in [.circleci/config.yml](https://github.com/bitovi/bitops/blob/master/.circleci/config.yml)

CircleCI console; `https://app.circleci.com/pipelines/github/bitovi/bitops?filter=all`

### LINKED ISSUE #162: Handle AWS SSO credentials
**Is your feature request related to a problem? Please describe.**
AWS SSO credentials require `AWS_SESSION_TOKEN` to work properly.

**Describe the solution you'd like**
Specify the `AWS_SESSION_TOKEN` when calling BitOps so bitops works with aws SSO credentials.

### PR DESCRIPTION
**Pull request by mickmcgrath13**
# Description

AWS SSO credentials require `AWS_SESSION_TOKEN` to work properly.  This PR adds the env var to the `/root/.aws/credentials` file if the env var exists.

(note: I would expect the env vars alone to work, but I tested that, and they do not)

Fixes #162
Fixes #141 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

1. Locally mounting to the bitops repo to run against an ops repo by modifying the main deploy script and running aws commands before `exit 0` before all the other code can run
2. Create a tagged bitops image and use in a pipeline runner

**Logs**
bitops command:
```
docker run --rm --name bitops \
-e AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
-e AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}" \
-e AWS_SESSION_TOKEN="${AWS_SESSION_TOKEN}" \
-e AWS_DEFAULT_REGION="${AWS_DEFAULT_REGION}" \
-e ENVIRONMENT="example" \
-e TF_STATE_BUCKET="bitovi-operations-devops-training-example" \
-e DEFAULT_FOLDER_NAME="_default" \
-v $(pwd):/opt/bitops_deployment \
-v /Users/mick/Projects/Bitovi/bitops:/opt/bitops \
bitovi/bitops:v1.0.13
```

aws command
```
###########################################
##### DEBUGGING
###########################################
aws s3api list-buckets
exit 0
###########################################
##### END DEBUGGING
###########################################
```

without change
```
An error occurred (InvalidAccessKeyId) when calling the ListBuckets operation: The AWS Access Key Id you provided does not exist in our records.
```

With change
```
{
    "Buckets": [
        {
            "Name": "bitovi-operations-devops-training",
            "CreationDate": "2022-04-04T18:23:05+00:00"
        },
        {
            "Name": "bitovi-operations-devops-training-example",
            "CreationDate": "2022-04-04T21:03:18+00:00"
        }
    ]
}
```

**Test Configuration**:
* BitOps image tag used:
* Hardware:
* Tool:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by PhillypHenning** [comment]
👍 Great description of work. Thanks for keeping it clean. 

👍 Code looks good. I can't see any potential problems arising

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    # TODO: when we have tests
+    # - run: |
+    #   ./scripts/ci/install.sh
+    #   ./scripts/ci/test.sh
+    - name: Publish Docker
```
Comment (PhillypHenning) [inline .github/workflows/build-and-publish.yml:20]: If I was pushed to name one thing to improve, I'd maybe request that the name `Publish Docker` be changed to `Publish to Docker` or `Publish Docker image`

File: .github/workflows/publish-docs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Publish Docs
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - "*"
+    paths:
+      - "docs/**"
```
Comment (ConnorGraham) [inline .github/workflows/publish-docs.yml:9]: watching for changes to `mkdocs.yml` would be good too

**Review by ConnorGraham — APPROVED** [review]

**Review by dylan-bitovi — APPROVED** [review]
👀 LGTM!

**Review by PhillypHenning — APPROVED** [review]

File: .github/workflows/publish-docs.yml (hunk 2 above)
Comment (mickmcgrath13) [reply .github/workflows/publish-docs.yml:9]: done

**Pull request merged**